source/_posts/2017-08-11-How-we-debug-with-Charles.md

```diff
@@ -0,0 +1,103 @@
+
+* When the app is making network requests to an API or a third party provider and you are not receiving the expected results in the app.
+* The app displays an empty view, when the app is supposed to fetch data from the API – you need to make sure to check whether data is received correctly from the API.
+* If you are making changes in the app and the changes are not applied – you need to check if the app is sending the data correctly or whether you are receiving the processed data from the server unexpectedly.
+* Any other scenario that involves the app making or receiving network calls – you need to check your Charles!
+
+With Charles, we are not only able to monitor the implementation of API, we can also monitor the implementation of third party providers such as push providers, Google Analytics, Hockey and furthermore we can monitor the order in which they are implemented. To ensure an agile workflow and a quick response to the issues that are found. Charles sessions can be exported and shared across the team. This 'comes in handy' when a tester is discovering an issue and wants to show the developer what is happening without the aforementioned steps to reproduce the issue. By exporting the session from Charles and adding it as an attachment in the ticket, it becomes very easy for the developer to reproduce the problem. Charles also has a built in way to publish gists to Github. The gist will automatically contain all required information in order to reproduce the issue such as auth token, request headers, API path, API environment, etc, etc.. This makes the workflow around discovering and reporting bugs in API or API implementation very quick and agile.
+
+The attached example beneath, is a screenshot of a Charles session, in which I am recording the traffic in the Apple Store App for iOS. On app open, we can for example see that the app calls a Content Delivery Network to get a feature graphic of an iPhone X.
+
+For testers the recording tool in Charles is the most important tool in the Charles toolbox. It gives us an insight into how mobile applications are built and furthermore allows us to identify issues on a lower level than otherwise possible.
+
```

Comment:
> lower level than otherwise possible.

"than otherwise possible" doesn't really make sense to me. You could just leave it as "on a lower level."